app/src/main/res/layout/admin_pin_activity.xml

```diff
@@ -0,0 +1,78 @@
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  xmlns:profile="http://schemas.android.com/tools">
+  <data>
+    <variable
+      name="viewModel"
+      type="org.oppia.app.profile.AdminPinViewModel" />
+  </data>
+  <ScrollView
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <androidx.constraintlayout.widget.ConstraintLayout
+      android:id="@+id/admin_auth_container"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:background="@color/addProfileBackground">
+      <TextView
+        android:id="@+id/admin_pin_main_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/admin_pin_requirement_description"
+        android:textColor="@color/oppiaPrimaryText"
+        android:textSize="16sp"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:layout_marginTop="32dp"
+        android:paddingStart="32dp"
+        android:paddingEnd="32dp"/>
+      <TextView
+        android:id="@+id/admin_pin_warning_text"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textSize="16sp"
+        android:textColor="@color/oppiaPrimaryText"
+        app:layout_constraintTop_toBottomOf="@id/admin_pin_main_text"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:layout_marginTop="16dp"
+        android:paddingStart="32dp"
+        android:paddingEnd="32dp"/>
+      <org.oppia.app.profile.ProfileInputView
+        android:id="@+id/input_pin"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        app:layout_constraintTop_toBottomOf="@id/admin_pin_warning_text"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:label="@string/admin_pin_new_pin"
+        app:isPasswordInput="true"
+        app:inputLength="5"
+        profile:error="@{viewModel.pinErrorMsg}"/>
+      <org.oppia.app.profile.ProfileInputView
+        android:id="@+id/input_confirm_pin"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/input_pin"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:label="@string/admin_pin_new_confirm"
+        app:isPasswordInput="true"
+        app:inputLength="5"
+        profile:error="@{viewModel.confirmPinErrorMsg}"/>
+      <Button
+        android:id="@+id/submit_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/state_continue_submit"
```

Comment:
I don't think we should reuse this string--this string is for the state player. Translators may choose to use a different word in this context.

Reply:
Updated to use its own string resource.